File: algorithms/coding-interview-questions-blind75-lol/1/best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {

        const int N = size(prices);

        int max_profit = 0;
        int mi = prices[0];

        for (int i = 1; i < N; ++i) {
            max_profit = max(max_profit, prices[i] - mi);
            mi = min(mi, prices[i]);
        }

        return max_profit;
    }
};

File: algorithms/blind75/1/best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        

        max_profit = 0
        mi = float('+inf')
        for i in range(len(prices)):
            if mi > prices[i]:
                mi = prices[i]
            elif prices[i] - mi > max_profit:
                max_profit = prices[i] - mi

        
        return max_profit
